Task2.3/Classes/Trainig.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task2._3.Classes
+{
+    public class Trainig
+    {
+        string? Description;
+        public Lesson[] Lessons { get; private set; } = new Lesson[2];
+        int _lessonsCount = 0;
+
+        public void Add(Lesson lesson)
+        {
+            if (_lessonsCount == Lessons.Length - 1)
+            {
+                var newLessons = new Lesson[Lessons.Length * 2];
+                Lessons.CopyTo(newLessons, 0);
+                Lessons = newLessons;
+            }
+            Lessons[_lessonsCount++] = lesson;
+        }
+
+        public bool IsPractical()
+        {
+            if (_lessonsCount == 0)
+            {
+                return false;
+            }
+
+            bool isPractical = true;
+
+            for (int i = 0; i < _lessonsCount; i++)
+            {
+                if (Lessons[i] is not PracticalLesson)
+                {
+                    isPractical = false;
+                    break;
+                }
+            }
+            return isPractical;
+        }
+
+        public void Clone(Trainig trainingToCopy)
```

Comment:
Clone() has to create object to clone to itself, not takes an empty object as parameter-> no guarantee that this object will be empty